signbank/dictionary/csv_import.py

```diff
@@ -0,0 +1,705 @@
+    request.session["validation_records"] = validation_records
+    request.session["question_numbers"] = question_numbers
+    request.session["question_gloss_map"] = question_to_gloss_map
+
+    return render(request, "dictionary/import_qualitrics_csv_confirmation.html",
+                  {"validation_records": validation_records, "skipped_rows": skipped_rows})
+
+
+@login_required
+@permission_required("dictionary.import_csv")
+@transaction.atomic()
+def confirm_import_qualitrics_csv(request):
+    """This view adds the data to database if the user confirms the action"""
+    if not request.method == "POST":
+        # If request method is not POST, redirect to the import form
+        return HttpResponseRedirect(reverse("dictionary:import_qualitrics_csv"))
+
+    if "cancel" in request.POST:
+        # If user cancels adding data, flush session variables
+        request.session.pop("validation_records", None)
+        request.session.pop("question_numbers", None)
+        request.session.pop("question_gloss_map", None)
+        # Set a message to be shown so that the user knows what is going on.
+        messages.add_message(request, messages.WARNING, _("Cancelled adding CSV data."))
+        return HttpResponseRedirect(reverse("dictionary:import_qualitrics_csv"))
+
+    elif "confirm" in request.POST:
+        validation_records_added = []
+        gloss_pk_list = []
+        validation_records = []
+
+        if "validation_records" and "question_numbers" and "question_gloss_map" in request.session:
+            # Retrieve glosses
+            gloss_pk_list = request.session["question_gloss_map"].values()
+            glosses = Gloss.objects.filter(pk__in=gloss_pk_list)
+            gloss_dict = {gloss.pk: gloss for gloss in glosses}
+
+            questions_numbers = request.session["question_numbers"]
+            question_gloss_map = request.session["question_gloss_map"]
+            validation_records = request.session["validation_records"]
+
+            # Go through csv data
+            for record in validation_records:
+                response_id = record.get("ResponseId", "")
+                respondent_first_name = record.get("RecipientFirstName", "")
+                respondent_last_name = record.get("RecipientLastName", "")
+                respondent_email = record.get("RecipientEmail", "")
+
+                for question_number in questions_numbers:
+                    contact_nzsl = False
+                    comment_written = record.get(f"{question_number}_Q2", None)
+                    if comment_written:
+                        comment_written = comment_written.split(",")
+                        if len(comment_written) > 1:
+                            contact_nzsl = True
+
+                    sign_seen = (record[f"{question_number}_Q1_1"]).lower()
+                    if sign_seen == "not sure ":
+                        sign_seen = "not_sure"
```

Comment:
**suggestion**: use `ValidationRecord.SignSeenChoices.NOT_SURE.value` to avoid duplicate string literals